custom-recipes/ocr-text-extraction-dataset/recipe.json

```diff
@@ -31,25 +32,42 @@
             "acceptsDataset": true
         }
     ],
-
+    "paramsPythonSetup": "select_ocr_engine.py",
     "params": [
         {
             "name": "recombine_pdf",
             "label" : "Recombine multiple-page PDF together",
             "type": "BOOLEAN",
-            "description": "Text of images that are from the same original multiple-page PDF (images with name pattern <PDF_NAME>_pdf_page_XXXXX.jpg) are concatenated."
+            "description": "Multiple-page PDFs and images with name pattern $FILENAME_pdf_page_XXXXX.jpg are extracted into a single row."
+        },
+        {
+            "name": "ocr_engine",
+            "label": "OCR Engine",
+            "type": "SELECT",
+            "mandatory": true,
+            "description": "",
+            "defaultValue": "default",
+            "getChoicesFromPython": true
         },
         {
             "name": "advanced_parameters",
             "label" : "Advanced preprocessing parameters",
-            "type": "BOOLEAN"
+            "type": "BOOLEAN",
+            "visibilityCondition" : "model.ocr_engine != 'default'"
```

Comment:
We cannot have a language param for the default engine because we don't know which param to display between "language" and "language_easyocr" when default is selected.
And it also makes sense not to have advanced params when users keep the default engine.

Reply:
We should add some help message though, to encourage user to select an engine explicitly if they want to modify advanced parameters

Reply:
there is no clean way to have help message in a plugin recipe UI and IMO hiding the param by default makes the cleanest UI